scripts/rddepman.ts

```diff
@@ -0,0 +1,197 @@
+
+    return;
+  }
+
+  // load current versions of dependencies
+  const depVersionsPath = path.join('src', 'assets', 'dependencies.yaml');
+  const currentVersions = await readDependencyVersions(depVersionsPath);
+
+  // get a list of dependencies' version comparisons
+  const versionComparisons: VersionComparison[] = await Promise.all(dependencies.map(async(dependency) => {
+    return {
+      name:           dependency.name,
+      currentVersion: currentVersions[dependency.name as keyof DependencyVersions],
+      latestVersion:  await dependency.getLatestVersion(),
+    };
+  }));
+
+  // filter comparisons down to the ones that need an update
+  const updatesAvailable = versionComparisons.filter(({ name, currentVersion, latestVersion }) => {
+    const equal = compareVersions(currentVersion, latestVersion);
+
+    if (equal) {
+      console.log(`${ name } is up to date.`);
+    } else {
+      console.log(`Can update ${ name } from ${ currentVersion } to ${ latestVersion }`);
+    }
+
+    return !equal;
+  });
+
+  // reconcile dependencies that need an update with state of repo's PRs
+  const needToCreatePR: VersionComparison[] = [];
+
+  await Promise.all(updatesAvailable.map(async({ name, currentVersion, latestVersion }) => {
+    // try to find PR for this combo of name, current version and latest version
+    const branchName = getBranchName(name);
+
+    const prs = await getPulls({
+      owner: GITHUB_OWNER, repo: GITHUB_REPO, head: `${ GITHUB_OWNER }:${ branchName }`, state: 'all', per_page: 100,
+    });
+
+    const title = getTitle(name, currentVersion, latestVersion);
+    let prExists = false;
+
+    await Promise.all(prs.map(async(pr) => {
+      if (pr.title !== title && pr.state === 'open') {
+        console.log(`Closing stale PR "${ pr.title }".`);
+        await getOctokit().rest.pulls.update({
+          owner: GITHUB_OWNER, repo: GITHUB_REPO, pull_number: pr.number, state: 'closed',
+        });
+      } else if (pr.title === title) {
+        console.log(`Found existing PR "${ title }".`);
+        prExists = true;
+      }
+    }));
+    if (!prExists) {
+      console.log(`Could not find PR "${ title }". Will create.`);
+      needToCreatePR.push({
+        name, currentVersion, latestVersion,
+      });
+    }
+  }));
+
+  // create a branch for each version update, make changes, and make a PR from the branch
+  for (const { name, currentVersion, latestVersion } of needToCreatePR) {
+    const branchName = getBranchName(name);
+    const commitMessage = `Bump ${ name } from ${ currentVersion } to ${ latestVersion }`;
+
+    git('checkout', '-b', branchName, MAIN_BRANCH);
+    const depVersions = await readDependencyVersions(depVersionsPath);
+
+    depVersions[name as keyof DependencyVersions] = latestVersion as string & AlpineLimaISOVersion;
+    await writeDependencyVersions(depVersionsPath, depVersions);
+    git('add', '.');
```

Comment:
Rather than adding everything, it might be better to explicitly commit the versions YAML file?